### ISSUE TITLE
[minor] FFmpeg blocks node from exiting

### ISSUE THREAD

**Issue by OnkelTem** [amishshah/prism-media#71]
It seems like basic creation of a `prism.FFmpeg` stream makes node to think it should wait.

```ts 
// ...
const transcoderStream = new prism.FFmpeg({
   args: ['-i', '-', '-f', 's16le', '-ar', '16000', '-channel_layout', 'mono'],
});
// ...
```

When launched, it's continuing to wait upon finishing everything else. 

Running it with wtfnode:
```
$ wtfnode script.js
```
results to:
```
...
[WTF Node?] open handles:
- File descriptors: (note: stdio always exists)
  - fd 2 (tty) (stdio)
  - fd 1 (tty) (stdio)
- Child processes
  - PID 24331
    - Entry point: /projects/custom/ts-zello/node_modules/prism-media/src/core/FFmpeg.js:156
    - STDIO file descriptors: 22, 24, 26
```

**Comment by amishshah** [comment]
Yep, the stream spawns an FFmpeg process. To fix this, you should make sure to call `stream.destroy()` when you no longer need the stream.

**Comment by OnkelTem** [comment]
Thanks Amish. Got it.

But isn't it possible to just do like:
`process.on('beforeExit', () => stream.destroy())` _inside_ Prism media? (Or 'exit', not sure about the difference)

**Comment by amishshah** [comment]
I wasn't aware this was a thing! I'll take a look at it. I still think it's best practice that the developer makes sure to clean up these things by themselves. However, I can't think of a strong reason against including this, so I'll try and add this to the TypeScript refactor if it works well.

**Comment by OnkelTem** [comment]
Cool!

Amish, unrelated. Do you hang out somewhere like Discord/Gitter/Slack whatever? Would be great to have an auxiliary communication channel with you! :)

**Comment by amishshah** [comment]
I'm available on Discord :) - hydrabolt#0001
